reference-implementation/to-upstream-wpts/transform-streams/general.js

```diff
@@ -300,4 +300,42 @@ promise_test(() => {
   return readableStreamToArray(ts.readable);
 }, 'methods should not not have .apply() or .call() called');
 
+async_test(t => {
+  let startCalled = false;
+  let startDone = false;
+  let transformDone = false;
+  let flushDone = false;
+  const ts = new TransformStream({
+    start() {
+      startCalled = true;
+      return new Promise(resolve => setTimeout(resolve, 90))
+        .then(() => { startDone = true; });
+    },
+    transform() {
+      t.step(() => {
+        assert_true(startDone, 'startPromise must resolve before transform is called');
+      });
+      return new Promise(resolve => setTimeout(resolve, 30))
+          .then(() => { transformDone = true; });
```

Comment:
I'm just putting everything inside .step() when possible so that any mistake can be captured by the testharness framework. It looks we can here, maybe.